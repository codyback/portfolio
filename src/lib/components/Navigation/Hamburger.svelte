<script>
	export let open = false;
</script>

<button
	class="sm:hidden flex items-center justify-items-end m-auto h-10 w-fit hover:text-gray-700 cursor-pointer border-none focus:outline-none"
	class:open
	on:click={() => (open = !open)}
>
	<svg
		width="28"
		height="24"
	>
		<line
			id="top"
			x1="0"
			y1="3"
			x2="28"
			y2="3"
		/>
		<line
			id="middle"
			x1="0"
			y1="11"
			x2="28"
			y2="11"
		/>
		<line
			id="bottom"
			x1="0"
			y1="19"
			x2="28"
			y2="19"
		/>
	</svg>
</button>

<style>
	svg {
		min-height: 24px;
		transition: transform 0.3s ease-in-out;
	}

	svg line {
		stroke: currentColor;
		stroke-width: 3;
		transition: transform 0.3s ease-in-out;
	}

	button {
		z-index: 20;
	}

	.open svg {
		transform: scale(0.9);
	}

	.open #top {
		transform: translate(5px, 0px) rotate(45deg);
	}

	.open #middle {
		opacity: 0;
	}

	.open #bottom {
		transform: translate(-11px, 9px) rotate(-45deg);
	}
</style>
