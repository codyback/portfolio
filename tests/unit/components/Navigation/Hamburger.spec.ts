import Hamburger from '@/lib/components/Navigation/Hamburger.svelte';
import { render } from '@testing-library/svelte';

describe('Hamburger', () => {
	it('renders correctly with open false', () => {
		const { container } = render(Hamburger, { open: false });

		expect(container).toMatchSnapshot();
	});

	it('renders correctly with open true', () => {
		const { container } = render(Hamburger, { open: true });

		expect(container).toMatchSnapshot();
	});
});
