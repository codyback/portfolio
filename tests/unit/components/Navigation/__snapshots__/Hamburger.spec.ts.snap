// Vitest Snapshot v1

exports[`Hamburger > renders correctly with open false 1`] = `
<body>
  <div>
    <button
      class="sm:hidden flex items-center justify-items-end m-auto h-10 w-fit hover:text-gray-700 cursor-pointer border-none focus:outline-none s-zPZ177rNd-sa"
    >
      <svg
        class="s-zPZ177rNd-sa"
        height="24"
        width="28"
      >
        <line
          class="s-zPZ177rNd-sa"
          id="top"
          x1="0"
          x2="28"
          y1="3"
          y2="3"
        />
        <line
          class="s-zPZ177rNd-sa"
          id="middle"
          x1="0"
          x2="28"
          y1="11"
          y2="11"
        />
        <line
          class="s-zPZ177rNd-sa"
          id="bottom"
          x1="0"
          x2="28"
          y1="19"
          y2="19"
        />
      </svg>
    </button>
    <!--&lt;Hamburger&gt;-->
  </div>
</body>
`;

exports[`Hamburger > renders correctly with open true 1`] = `
<body>
  <div>
    <button
      class="sm:hidden flex items-center justify-items-end m-auto h-10 w-fit hover:text-gray-700 cursor-pointer border-none focus:outline-none s-zPZ177rNd-sa open"
    >
      <svg
        class="s-zPZ177rNd-sa"
        height="24"
        width="28"
      >
        <line
          class="s-zPZ177rNd-sa"
          id="top"
          x1="0"
          x2="28"
          y1="3"
          y2="3"
        />
        <line
          class="s-zPZ177rNd-sa"
          id="middle"
          x1="0"
          x2="28"
          y1="11"
          y2="11"
        />
        <line
          class="s-zPZ177rNd-sa"
          id="bottom"
          x1="0"
          x2="28"
          y1="19"
          y2="19"
        />
      </svg>
    </button>
    <!--&lt;Hamburger&gt;-->
  </div>
</body>
`;
